### PR TITLE
LocalizationStreamParser should not fail on faulty PO files #7387

### PR DESCRIPTION
**Pull request by MpDzik**

### CODE REVIEW

File: src/Orchard/Localization/Services/LocalizationStreamParser.cs [hunk 1]
```diff
@@ -96,27 +103,33 @@ public class LocalizationStreamParser : ILocalizationStreamParser {
             return sb == null ? str : sb.ToString();
         }
 
-        private static string TrimQuote(string str) {
```
Comment (sebastienros) [inline src/Orchard/Localization/Services/LocalizationStreamParser.cs]: Why did you remove the `static`? It's not better without it.

Comment (MpDzik) [inline src/Orchard/Localization/Services/LocalizationStreamParser.cs]: Because I need to access the `Logger` property (which is non-static) to write a warning to the log file. Alternatively, I can leave out the warning and just interpret this single quote as an empty string. I'm not sure which solution is better.

Comment (sebastienros) [inline src/Orchard/Localization/Services/LocalizationStreamParser.cs]: I see, that's fine then,
